packages/material-ui/src/Link/Link.js

```diff
@@ -1,62 +1,100 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { elementTypeAcceptingRef } from '@material-ui/utils';
+import { deepmerge, elementTypeAcceptingRef } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import capitalize from '../utils/capitalize';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import useIsFocusVisible from '../utils/useIsFocusVisible';
 import useForkRef from '../utils/useForkRef';
 import Typography from '../Typography';
+import linkClasses, { getLinkUtilityClass } from './linkClasses';
 
-export const styles = {
-  /* Styles applied to the root element. */
-  root: {},
-  /* Styles applied to the root element if `underline="none"`. */
-  underlineNone: {
-    textDecoration: 'none',
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[`underline${capitalize(styleProps.underline)}`],
+    ...(styleProps.component === 'button' && styles.button),
+    ...(styleProps.focusVisible && styles.focusVisible),
```

Comment:
```suggestion
    ...(styleProps.focusVisible && {[`&.${linkClasses.focusVisible}`]: styles.focusVisible }),
```

Reply:
We are increasing the specificity for the pseudo classes, so we must do the same on the overrides.

Reply:
Apologies, I've edited the suggestion. We should not add overrides for the pseudo selectors, only classes. It was never supported.